### PR TITLE
DBZ-7162 Various streaming metrics for MongoDB connector

### PR DESCRIPTION
**Pull request by jcechace**
https://issues.redhat.com/browse/DBZ-7162

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/metrics/MongoDbStreamingChangeEventSourceMetrics.java [hunk 1]
```diff
@@ -44,6 +57,47 @@ public long getNumberOfDisconnects() {
...
+            timer = lastEmptyPollTime;
+        }
+        else {
+            counter = numberOfSourceEvents;
+            timer = lastSourceEventPollTime;
+        }
+
+        counter.incrementAndGet();
+        if (clock != null && prePollTimestamp != null) {
+            var now = clock.currentTimeAsInstant();
```
Comment (jpechane) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/metrics/MongoDbStreamingChangeEventSourceMetrics.java:95]: @jcechace This should be the first statement in this method.
When can `clock` be empty?

Comment (jcechace) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/metrics/MongoDbStreamingChangeEventSourceMetrics.java:95]: True.. When you don't provide them :) If you think that the guarding is pointless here I can remove it.

Comment (jcechace) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/metrics/MongoDbStreamingChangeEventSourceMetrics.java]: Updated. It should be OK now.

**Review by obabec — APPROVED** [review]

**Comment by jpechane** [comment]
@jcechace Applied, thanks

**Pull request merged**